src/Symfony/Component/Form/Extension/Core/Type/BaseType.php

```diff
@@ -79,6 +79,9 @@ public function buildView(FormView $view, FormInterface $form, array $options)
         for ($type = $form->getConfig()->getType(); null !== $type; $type = $type->getParent()) {
             array_unshift($blockPrefixes, $type->getBlockPrefix());
         }
+        if (null !== $options['block_prefix']) {
+            $blockPrefixes[] = $options['block_prefix'];
```

Comment:
Just asking: should we validate that this user-generated `block_prefix` only contains valid characters? After all this name is used indirectly as a PHP method name (see https://github.com/twigphp/Twig/blob/84b2eb541026dbd39590ba80269ae7676b1bfc46/lib/Twig/Node/Block.php#L29) so not all characters are valid. Thanks!

Reply:
Hi Javier! Thanks for your input. Actually, we don't do that for the rest of alternatives: `block_name` or `FormTypeInterface::getBlockPrefix()` and IMHO not worth it, if the user makes the mistake, currently he receive this error message:
![invalid_method_name](https://user-images.githubusercontent.com/2028198/50450797-1d6e6400-08fe-11e9-8196-7ba6b4ff2c5a.png)
Highlighting the creation of the form in exceptions stack.

Is it enough for you? Otherwise, we can validate all block prefixes to be more accurate and show a clear error message. I suppose that a separate report will be required if so. Cheers!